src/zc/relationship/README.txt

```diff
@@ -336,14 +371,112 @@ zope.interface.Method associated with an interface, typically obtained
 using a spelling like `IRelationship['subjects']`) or a dict.  Each dict
 must have either the 'element' key, which is the interface element to be
 indexed; or the 'callable' key, which is the callable shown in the
-simpler, introductory example above [#there_can_be_only_one]_.  It then
+simpler, introductory example above [#there_can_be_only_one]_.
+
+.. [#there_can_be_only_one] instantiating an index with a dictionary containing
+    both the 'element' and the 'callable' key is an error:
+
+    >>> def subjects(obj, index, cache):
+    ...     return obj.subjects
+    ...
+    >>> ix = index.Index(
+    ...     ({'element': IRelationship['subjects'],
+    ...       'callable': subjects, 'multiple': True},
+    ...      IRelationship['relationshiptype'],
+    ...      {'element': IRelationship['objects'], 'multiple': True},
+    ...      IContextAwareRelationship['getContext']),
+    ...     index.TransposingTransitiveQueriesFactory('subjects', 'objects'))
+    Traceback (most recent call last):
+    ...
+    ValueError: cannot provide both callable and element
+
+    While we're at it, as you might expect, you must provide one of them.
+
+    >>> ix = index.Index(
+    ...     ({'multiple': True},
+    ...      IRelationship['relationshiptype'],
+    ...      {'element': IRelationship['objects'], 'multiple': True},
+    ...      IContextAwareRelationship['getContext']),
+    ...     index.TransposingTransitiveQueriesFactory('subjects', 'objects'))
+    Traceback (most recent call last):
+    ...
+    ValueError: must provide element or callable
+
+It then
 can contain other keys to override the default indexing behavior for the
 element.
 
 The element's or callable's __name__ will be used to refer to this
 element in queries, unless the dict has a 'name' key, which must be a
 non-empty string [#name_errors]_.
 
+.. [#name_errors] It's possible to pass a callable without a name, in which
+    case you must explicitly specify a name.
+
+    >>> class AttrGetter(object):
+    ...     def __init__(self, attr):
+    ...         self.attr = attr
+    ...     def __lt__(self, other):
+    ...         return self.attr < getattr(other, 'attr', other)
```

Comment:
I don't like seeing only one of six rich comparison methods being defined. If you use @functools.total_ordering, you can get away by defining two of six.